### ISSUE TITLE
Solaris11: hcrypto test failures with 7.6.0

### ISSUE THREAD

**Issue by quanah** [heimdal/heimdal#564]
Building out 7.6.0, linked to OpenSSL 1.1.1b, I get the following test failures:

```
make[5]: Entering directory '/export/home/build/git/sold/sold-master/SunOS11.4-i86pc_64/build/heimdal.i86pc_64/lib/hcrypto'
PASS: destest
FAIL: mdtest
PASS: rc2test
PASS: rctest
PASS: test_bn
PASS: test_bulk
FAIL: test_cipher
PASS: test_engine_dso
FAIL: test_hmac
FAIL: test_pkcs12
FAIL: test_pkcs5
FAIL: test_crypto
============================================================================
Testsuite summary for Heimdal 7.6.0
============================================================================
# TOTAL: 12
# PASS:  6
# SKIP:  0
# XFAIL: 0
# FAIL:  6
# XPASS: 0
# ERROR: 0
============================================================================
See lib/hcrypto/test-suite.log
Please report to https://github.com/heimdal/heimdal/issues
============================================================================
```

Log file shows:
```
FAIL: mdtest
============

mdtest: rc4 8: EVP_CipherInit_ex encrypt
MD2... FAIL mdtest (exit status: 1)

FAIL: test_cipher
=================

test_cipher: rc4 8: 0 EVP_CipherInit_ex encrypt
rc2 not supported
rc2-40 not supported
FAIL test_cipher (exit status: 1)

FAIL: test_hmac
===============

test_hmac: rc4 8: EVP_CipherInit_ex encrypt
FAIL test_hmac (exit status: 1)

FAIL: test_pkcs12
=================

test_pkcs12: rc4 8: EVP_CipherInit_ex encrypt
FAIL test_pkcs12 (exit status: 1)

FAIL: test_pkcs5
================

test_pkcs5: rc4 8: EVP_CipherInit_ex encrypt
FAIL test_pkcs5 (exit status: 1)

FAIL: test_crypto
=================

....
test_rand: unknown method egd
random egd is not available
test_rand: unknown method w32crypto
random w32crypto is not available
example_evp_cipher: rc4 8: EVP_CipherInit_ex encrypt
1 failed
FAIL test_crypto (exit status: 1)
```

**Comment by quanah** [comment]
There seems to be something seriously wrong with the test suite code itself.  For example, the mdtest binary exits on this call:

```
if (hash->evp() == NULL) {
```

If I step through in GDB, I get the same behavior:

```
277             if (hash->evp() == NULL) {
(gdb) print hash->evp()
mdtest: rc4 8: EVP_CipherInit_ex encrypt
MD2... [Inferior 1 (process 4036    ) exited with code 01]
The program being debugged exited while in a function called from GDB.
Evaluation of the expression containing the function
(hc_EVP_md2@plt) will be abandoned.
(gdb)
```

**Comment by quanah** [comment]
After working with @jaltman today, there are two separate issues:

1) All the test failures outside of test_cipher are related to the hcrypto_validate function.  Turning it into a null op (I.e., set validated = 1 or increment it before the if (validated) statement) allows those tests to pass.  This change is of course not one that should be pushed, as that function needs to execute properly at runtime, so some change is necessary to make it so it also works with the test suite.

2) For test_cipher, this is due to the fact that pkcs11-rc4 is not initializing correctly.  I see a failure at line 330 of evp-pkcs11.c, where the return value is set to 112 (CKR_MECHANISM_INVALID).   This later causes test_cipher to exit with an error.  For now I can work around this by not testing pkcs11-rc4 in test_cipher, but really this should be recorded as the cipher not being available, rather than causing a test failure.

**Comment by quanah** [comment]
Fixed by 5c70e5015edbe056eb192c166c3056c29d70482b
